### PR TITLE
Mybudget save to mongo

### PR DESCRIPTION
**Pull request by jcastiarena**
Handles a request with the form:
curl -v -XPOST localhost:8000/api/mybudget -d '[{"id":"558c66816d2919e68e36543d","amount":"5.0"}, {"id":"558c66816d2919e68e36543c","amount":"10.5"}, {"id":"558c66816d2919e68e36543b","amount":"2.5"}]' -H 'Content-Type: application/json'

Creates a MyBudget object from the posted json as an array of rows of type MyBudgetItem and persists it to mongodb.

### CODE REVIEW

File: lib/data/models/mybudget.js [hunk 1]
```diff
@@ -4,10 +4,11 @@ import MyBudgetItem from './mybudget-item'
 
 const log = debug('presupuesto:data:models:mybudget')
 let Schema = mongoose.Schema;
-let ObjectId = Schema.ObjectId;
+//let ObjectId = Schema.ObjectId;
 
 let MyBudgetSchema = new Schema({
-	rows: { type: [MyBudgetItem] },
+	rows: { type: [Schema.Types.Mixed] },
```
Comment (jfresco) [inline lib/data/models/mybudget.js:10]: Why `rows` is not of type `[MyBudgetItem]`?

File: lib/api/mybudget.js [hunk 2]
```diff
@@ -8,13 +8,14 @@ let routes = express()
...
   log(`Posted My Budget: %j`, req.body)
+  
   MyBudget.create(req.body, (err, data) => {
     if (err) {
       log(`Error creating MyBudget: ${err.toString()}`)
       return handleError(res)
     }
 
-    return res.sendStatus(200)
+    return res.json({result: 'Data was successfully saved'})
```
Comment (jfresco) [inline lib/api/mybudget.js:18]: I think returning a `200` in a `POST` operation seems enough.

File: lib/data/api/category.js [hunk 3]
```diff
@@ -9,3 +9,31 @@ export function getAll (cb) {
     .find()
     .exec(cb)
 }
+
+export function getCategoryByName (categoryName, err, success) {
+  //console.log('Querying category: ', categoryName)
+  log(`Querying category %j`, categoryName)
+  
+  Category.findOne({ name: categoryName }, function (err, category) {
```
Comment (jfresco) [inline lib/data/api/category.js]: We can use arrow functions.

File: lib/data/api/mybudget.js [hunk 4]
```diff
@@ -1,13 +1,88 @@
...
+*  array of budgets.
+*/
+export function extractBudgets (budgets, cb) {
+	//console.log('Extracting category budgets from: ', budgets)
+	log(`Extracting category budgets from: %j`, budgets)
+	
+	for(var i = 0; i < budgets.length; i++) {
+    	let categoryBudget = budgets[i];
+    	let categoryId = categoryBudget.id
+    	let categoryAmount = new Number(parseFloat(categoryBudget.amount))
```
Comment (jfresco) [inline lib/data/api/mybudget.js]: Is the `Number(parseFloat(...` really necessary?

File: lib/data/models/mybudget.js (hunk 1 above)
Comment (jcastiarena) [inline lib/data/models/mybudget.js:10]: For some reason it was throwing a CastException in my case when using [MyBudgetItem].

**Pull request merged**